### ISSUE TITLE
Uninstall profile is missing

### ISSUE THREAD

**Issue by erral** [eea/eea.facetednavigation#171]
Although the uninstall profile is registered:

https://github.com/eea/eea.facetednavigation/blob/master/eea/facetednavigation/profiles.zcml#L36

There is no such directory on the profiles directory: https://github.com/eea/eea.facetednavigation/tree/master/eea/facetednavigation/profiles

So after uninstalling the package from the controlpanel and removing it from buildout, some traces are left behind, such as toolbar  menu entries and the Plone toolbar gets unusable:

```
2019-05-14 10:44:15 ERROR plone.app.viewletmanager Error while rendering viewlet-manager=plone.toolbar, viewlet=plone.contentviews
Traceback (most recent call last):
  File "/home/erral/downloads/eggs/plone.app.viewletmanager-2.0.12-py2.7.egg/plone/app/viewletmanager/manager.py", line 106, in render
    html.append(viewlet.render())
  File "/home/erral/downloads/eggs/plone.app.layout-2.8.3-py2.7.egg/plone/app/layout/viewlets/common.py", line 59, in render
    return self.index()
  File "/home/erral/downloads/eggs/Zope2-2.13.27-py2.7.egg/Products/Five/browser/pagetemplatefile.py", line 125, in __call__
    return self.im_func(im_self, *args, **kw)
  File "/home/erral/downloads/eggs/Zope2-2.13.27-py2.7.egg/Products/Five/browser/pagetemplatefile.py", line 59, in __call__
    sourceAnnotations=getattr(debug_flags, 'sourceAnnotations', 0),
  File "/home/erral/downloads/eggs/zope.pagetemplate-4.2.1-py2.7.egg/zope/pagetemplate/pagetemplate.py", line 137, in pt_render
    strictinsert=0, sourceAnnotations=sourceAnnotations
  File "/home/erral/downloads/eggs/five.pt-2.2.5-py2.7.egg/five/pt/engine.py", line 98, in __call__
    return self.template.render(**kwargs)
  File "/home/erral/downloads/eggs/z3c.pt-3.0.0a1-py2.7.egg/z3c/pt/pagetemplate.py", line 163, in render
    return base_renderer(**context)
  File "/home/erral/downloads/eggs/Chameleon-2.25-py2.7.egg/chameleon/zpt/template.py", line 261, in render
    return super(PageTemplate, self).render(**vars)
  File "/home/erral/downloads/eggs/Chameleon-2.25-py2.7.egg/chameleon/template.py", line 171, in render
    self._render(stream, econtext, rcontext)
  File "b5982d2a7776791f96cd17d5cad4cc0f.py", line 157, in render
  File "/home/erral/downloads/eggs/z3c.pt-3.0.0a1-py2.7.egg/z3c/pt/expressions.py", line 74, in render_content_provider
    return cp.render()
  File "/home/erral/downloads/eggs/plone.app.contentmenu-2.2.4-py2.7.egg/plone/app/contentmenu/view.py", line 18, in render
    return self.index()
  File "/home/erral/downloads/eggs/Zope2-2.13.27-py2.7.egg/Products/Five/browser/pagetemplatefile.py", line 125, in __call__
    return self.im_func(im_self, *args, **kw)
  File "/home/erral/downloads/eggs/Zope2-2.13.27-py2.7.egg/Products/Five/browser/pagetemplatefile.py", line 59, in __call__
    sourceAnnotations=getattr(debug_flags, 'sourceAnnotations', 0),
  File "/home/erral/downloads/eggs/zope.pagetemplate-4.2.1-py2.7.egg/zope/pagetemplate/pagetemplate.py", line 137, in pt_render
    strictinsert=0, sourceAnnotations=sourceAnnotations
  File "/home/erral/downloads/eggs/five.pt-2.2.5-py2.7.egg/five/pt/engine.py", line 98, in __call__
    return self.template.render(**kwargs)
  File "/home/erral/downloads/eggs/z3c.pt-3.0.0a1-py2.7.egg/z3c/pt/pagetemplate.py", line 163, in render
    return base_renderer(**context)
  File "/home/erral/downloads/eggs/Chameleon-2.25-py2.7.egg/chameleon/zpt/template.py", line 261, in render
    return super(PageTemplate, self).render(**vars)
  File "/home/erral/downloads/eggs/Chameleon-2.25-py2.7.egg/chameleon/template.py", line 191, in render
    raise_with_traceback(exc, tb)
  File "/home/erral/downloads/eggs/Chameleon-2.25-py2.7.egg/chameleon/template.py", line 171, in render
    self._render(stream, econtext, rcontext)
  File "bcb4e106ae6432f6c1c0088d108934fa.py", line 105, in render
  File "/home/erral/downloads/eggs/five.pt-2.2.5-py2.7.egg/five/pt/expressions.py", line 161, in __call__
    return base()
  File "/home/erral/downloads/eggs/plone.app.contentmenu-2.2.4-py2.7.egg/plone/app/contentmenu/view.py", line 27, in menu
    items = menu.getMenuItems(self.context, self.request)
  File "/home/erral/downloads/eggs/zope.browsermenu-4.2-py2.7.egg/zope/browsermenu/menu.py", line 86, in getMenuItems
    for index, order, title, item in result]
  File "/home/erral/downloads/eggs/zope.browsermenu-4.2-py2.7.egg/zope/browsermenu/menu.py", line 182, in getMenu
    return menu.getMenuItems(object, request)
  File "/home/erral/downloads/eggs/plone.app.contentmenu-2.2.4-py2.7.egg/plone/app/contentmenu/menu.py", line 100, in getMenuItems
    editActions = context_state.actions('object_buttons')
  File "/home/erral/downloads/eggs/plone.memoize-1.2.3-py2.7.egg/plone/memoize/view.py", line 48, in memogetter
    value = cache[key] = func(*args, **kwargs)
  File "/home/erral/downloads/eggs/plone.app.layout-2.8.3-py2.7.egg/plone/app/layout/globals/context.py", line 246, in actions
    max=max,
  File "/home/erral/downloads/eggs/Products.CMFPlone-5.1.5-py2.7.egg/Products/CMFPlone/ActionsTool.py", line 83, in listActionInfos
    if check_condition and not ai['available']:
  File "/home/erral/downloads/eggs/Products.CMFCore-2.2.12-py2.7.egg/Products/CMFCore/ActionInformation.py", line 206, in __getitem__
    value = self.data[key] = value(self._ec)
  File "/home/erral/downloads/eggs/Products.CMFCore-2.2.12-py2.7.egg/Products/CMFCore/Expression.py", line 47, in __call__
    res = compiled(econtext)
  File "/home/erral/downloads/eggs/zope.tales-3.5.3-py2.7.egg/zope/tales/expressions.py", line 217, in __call__
    return self._eval(econtext)
  File "/home/erral/downloads/eggs/Zope2-2.13.27-py2.7.egg/Products/PageTemplates/Expressions.py", line 147, in _eval
    ob = self._subexprs[-1](econtext)
  File "/home/erral/downloads/eggs/zope.tales-3.5.3-py2.7.egg/zope/tales/expressions.py", line 124, in _eval
    ob = self._traverser(ob, element, econtext)
  File "/home/erral/downloads/eggs/Zope2-2.13.27-py2.7.egg/Products/PageTemplates/Expressions.py", line 74, in boboAwareZopeTraverse
    object = object.restrictedTraverse(name)
  File "/home/erral/downloads/eggs/Zope2-2.13.27-py2.7.egg/OFS/Traversable.py", line 317, in restrictedTraverse
    return self.unrestrictedTraverse(path, default, restricted=True)
  File "/home/erral/downloads/eggs/Zope2-2.13.27-py2.7.egg/OFS/Traversable.py", line 300, in unrestrictedTraverse
    raise e
AttributeError: @@faceted_subtyper

 - Expression: "provider:plone.contentmenu"
 - Filename:   ... .8.3-py2.7.egg/plone/app/layout/viewlets/contentviews.pt
 - Location:   (line 9: col 32)
 - Source:     ... :replace="structure provider:plone.contentmenu" />
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Expression: "view/menu"
 - Filename:   ... enu-2.2.4-py2.7.egg/plone/app/contentmenu/contentmenu.pt
 - Location:   (line 3: col 17)
 - Source:     define="menu view/menu;"
                            ^^^^^^^^^
 - Arguments:  repeat: {...} (0)
               template: <ViewPageTemplateFile - at 0x7f3959103ed0>
               views: <ViewMapper - at 0x7f395905d510>
               modules: <instance - at 0x7f39671ee908>
               args: <tuple - at 0x7f39705e6050>
               here: <ImplicitAcquisitionWrapper administrazioa-eta-kudeaketa at 0x7f39610f9fa0>
               user: <ImplicitAcquisitionWrapper - at 0x7f39610ef780>
               nothing: <NoneType - at 0x55e3e867e560>
               container: <ImplicitAcquisitionWrapper administrazioa-eta-kudeaketa at 0x7f39610f9fa0>
               request: <instance - at 0x7f3960da5b48>
               wrapped_repeat: <SafeMapping - at 0x7f39605ac368>
               traverse_subpath: <list - at 0x7f395929d7e8>
               default: <object - at 0x7f3970505560>
               loop: {...} (0)
               context: <ImplicitAcquisitionWrapper administrazioa-eta-kudeaketa at 0x7f39610f9fa0>
               view: <ContentMenuProvider plone.contentmenu at 0x7f3959103e50>
               translate: <function translate at 0x7f395aad5ed8>
               root: <ImplicitAcquisitionWrapper Zope at 0x7f396134a8c0>
               options: {...} (0)
               target_language: <NoneType - at 0x55e3e867e560>

```

Version information

- Plone 5.1.5 (5115)
- CMF 2.2.12
- Zope 2.13.27
- Python 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
- PIL 5.2.0 (Pillow)

**Comment by avoinea** [comment]
Also a custom cleanup script is needed if you want to completely remove the egg from your buildout as marker interface **IFacetedNavigable** and **__annotations__** will remain assigned to objects.

**Comment by erral** [comment]
This could be achieved using a custom import step or a run_after handler: https://docs.plone.org/develop/addons/components/genericsetup.html#custom-installer-code-setuphandlers-py

**Comment by erral** [comment]
Fixed in #173
